### PR TITLE
Remove unused/redundant code from the MigrationsSqlGenerators

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

File: src/EFCore.SqlServer/breakingchanges.netcore.json [hunk 1]
```diff
@@ -2,5 +2,15 @@
...
+      "Kind": "Removal"
+    },
+    {
+      "TypeId": "public class Microsoft.EntityFrameworkCore.Migrations.SqlServerMigrationsSqlGenerator : Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator",
+      "MemberId": "protected virtual System.Void ColumnDefinition(System.String schema, System.String table, System.String name, System.Type clrType, System.String type, System.Nullable<System.Boolean> unicode, System.Nullable<System.Int32> maxLength, System.Boolean rowVersion, System.Boolean nullable, System.Object defaultValue, System.String defaultValueSql, System.String computedColumnSql, System.Boolean identity, Microsoft.EntityFrameworkCore.Infrastructure.IAnnotatable annotatable, Microsoft.EntityFrameworkCore.Metadata.IModel model, Microsoft.EntityFrameworkCore.Migrations.MigrationCommandListBuilder builder)",
+      "Kind": "Removal"
```
Comment (bricelam) [inline src/EFCore.SqlServer/breakingchanges.netcore.json]: Can we before 3.0?

Comment (AndriySvyryd) [reply src/EFCore.SqlServer/breakingchanges.netcore.json]: It's provider code.
@ajcvickers

Comment (ajcvickers) [reply src/EFCore.SqlServer/breakingchanges.netcore.json]: I believe it is expected that applications may use SqlServerMigrationsSqlGenerator, which is why it is explicitly not in an internal namespace.

Comment (AndriySvyryd) [reply src/EFCore.SqlServer/breakingchanges.netcore.json]: Ok, I'll put an Obsolete on it then

**Pull request merged**